### PR TITLE
Use sigs.k8s.io/yaml/goyaml.v3

### PR DESCRIPTION
**Pull request by mkumatag**
Fixes: #33971

### CODE REVIEW

**Comment by aojea** [comment]
/lgtm
/approve

Thanks

File: go.mod [hunk 1]
```diff
@@ -75,8 +75,8 @@ require (
 	gopkg.in/fsnotify.v1 v1.4.7 // indirect
 	gopkg.in/ini.v1 v1.62.0 // indirect
 	gopkg.in/robfig/cron.v2 v2.0.0-20150107220207-be2e0b0deed5 // indirect
-	gopkg.in/yaml.v2 v2.4.0
-	gopkg.in/yaml.v3 v3.0.1
+	gopkg.in/yaml.v2 v2.4.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
```
Comment (BenTheElder) [inline go.mod:79]: do we know why we still have indirect dependencies?

**Review by BenTheElder — COMMENTED** [review]
thanks!

File: go.mod (hunk 1 above)
Comment (BenTheElder) [reply go.mod:79]: > Fixes: https://github.com/kubernetes/test-infra/issues/33971

Let's keep the issue open until it's dropped from go.mod

**Pull request merged**